src/action.ts

```diff
@@ -46,23 +46,34 @@ export async function waitForDeviceToComeOnline(
 
   return new Promise<boolean>((resolve, reject) => {
     const flashTimeout = setTimeout(() => {
-      stream.abort()
-      stream.stopIdleTimeout()
+      try {
+        stream.abort()
+        stream.stopIdleTimeout()
+      } catch (cleanupError) {
+        core.warning(`Error during stream cleanup: ${cleanupError}`)
+      }
       reject(new Error('timed out waiting for device to come back online'))
     }, timeoutMs)
 
     core.info('waiting for device to come online')
-    stream.on('event', (event: {data: string}) => {
+
+    stream.on('event', async (event: {data: string}) => {
       try {
         if (event.data === 'online') {
           core.info('device is online')
           clearTimeout(flashTimeout)
-          stream.abort()
-          stream.stopIdleTimeout()
+          try {
+            stream.abort()
+            stream.stopIdleTimeout()
```

Comment:
Might be better to call `stream.end()` instead of `stopIdleTimeout`?
This one also does the `stopIdleTimeout()` and other stuff.
https://github.com/particle-iot/particle-api-js/blob/e5535d1fe988dfa5e53c990b953ed88871d90260/src/EventStream.js#L125

Reply:
ah, good catch. That seems like a better way. I switched it over.